icu4c/source/common/locid.cpp

```diff
@@ -1576,10 +1576,8 @@ AliasReplacer::replaceTransformedExtensions(
             }
             tkey = nextTKey;
         } while (tkey != nullptr);
-        tfields.sort([](UElement e1, UElement e2) -> int8_t {
-            // uprv_strcmp return int and in some platform, such as arm64-v8a,
-            // it may return positive values > 127 which cause the casted value
-            // of int8_t negative.
+        tfields.sort([](UElement e1, UElement e2) -> int32_t {
+            // uprv_strcmp may return value other than {-1, 0, 1}.
             int res = uprv_strcmp(
                 (const char*)e1.pointer, (const char*)e2.pointer);
             return (res == 0) ? 0 : ((res > 0) ? 1 : -1);
```

Comment:
```suggestion
            return uprv_strcmp((const char*)e1.pointer, (const char*)e2.pointer);
```
I am pretty sure that strcmp() at worst returns the difference between unequal bytes, which means that it needs 9 bits, which easily fit into an int32_t. And usually an int is the same as an int32_t.

Reply:
but that does not mean it will return 0, -1 and 1 which is still mandated by UElementComparator

Reply:
Yes, see my [ticket comment from March 5](https://unicode-org.atlassian.net/browse/ICU-21526?focusedCommentId=160438). Please take a look at the UElementComparator call sites, I bet that few or none rely on these three values.

Reply:
I understand but icu4c/source/common/uelement.h still state

```
 * @return 0 if the two elements are equal, -1 if e1 is < e2, or +1 if e1 is > e2.
```

do you want me to change it to
```
 * @return 0 if the two elements are equal, a negative value  if e1 is < e2, or a positive value if e1 is > e2.
```
? If not changing there, then we cannot drop this change here, right?

Reply:
Please look at the call sites. If they don't actually need the return value to be one of these three values, or if they are easy to fix, then please change the "API" docs.